seisflows/workflow/inversion.py

```diff
@@ -224,10 +225,26 @@ def evaluate_objective_function(self, save_residuals=False, **kwargs):
         logger.debug(f"quantifying misfit with "
                      f"'{self.preprocess.__class__.__name__}'")
 
+        # If line search, add step count as suffix in the residuals file
```

Comment:
I think we can skip the second `if` call here and just evaluate all string formatters. If in the input string formatter does not have an `{sc}` then the `sc` input will just be skipped over. i.e.,

```python
        if save_residuals:
            save_residuals = save_residuals.format(src=self.solver.source_name,
                                                   it=str(self.iteration),
                                                   sc=str(self.optimize.step_count + 1))

```